message/messaging_service.cc

```diff
@@ -430,13 +430,26 @@ messaging_service::messaging_service(gms::feature_service& feature_service, conf
 {
     _rpc->set_logger(&rpc_logger);
 
+    std::optional<unsigned> maintenance_tenant_idx;
+
     // this initialization should be done before any handler registration
     // this is because register_handler calls to: scheduling_group_for_verb
     // which in turn relies on _connection_index_for_tenant to be initialized.
     _connection_index_for_tenant.reserve(_scheduling_config.statement_tenants.size());
     for (unsigned i = 0; i <  _scheduling_config.statement_tenants.size(); ++i) {
         auto& tenant_cfg = _scheduling_config.statement_tenants[i];
         _connection_index_for_tenant.push_back({tenant_cfg.sched_group, i, tenant_cfg.enabled});
+        
+        if (!_scheduling_config.statement_tenants[i].enabled && _scheduling_config.statement_tenants[i].name == "$maintenance") {
+            maintenance_tenant_idx = i;
+        }
+    }
+
+    if (maintenance_tenant_idx) {
+        _maintenance_tenant_enabled_listener = feature_service.maintenance_tenant.when_enabled([this, idx = std::move(*maintenance_tenant_idx)] {
+            _scheduling_config.statement_tenants[idx].enabled = true;
+            _connection_index_for_tenant[idx].enabled = true;
+        });
```

Comment:
Can we do this in main somehow? Then we don't need the previous patch and the hard-coding happens at the place where we supply the had-coded config anyway?
We just need a mutable `get_tenant_config()` for this, but I think that is the lesser of evils.